### PR TITLE
fix(appconnect): Add app_id_str column to appconnectbuild

### PR DESCRIPTION
**Pull request by loewenheim**
This is step 1 in the effort to change the type of `AppConnectBuild.app_id` from an integer to a string.

It adds a new column `app_id_str` of type `CharField` to the `appconnectbuild` model and a corresponding field to `AppConnectBuild`.

### CODE REVIEW

**Review by wedamija — COMMENTED** [review]
This pr should include the change that adds the field to the model. 

I noticed in https://github.com/getsentry/sentry/pull/40264/ that you are setting a default of "0". Since we're on pg14 you can actually create the column with a default if you like. https://develop.sentry.dev/database-migrations/#adding-columns-with-a-default. This also allows you to make the column not null, if that's helpful

**Comment by loewenheim** [comment]
Thank you for the feedback. I've now consolidated the two PRs and adjusted the migration to hopefully make it default-safe.

File: src/sentry/models/appconnectbuilds.py [hunk 1]
```diff
@@ -28,11 +28,11 @@ class AppConnectBuild(Model):
     project = FlexibleForeignKey("sentry.Project", db_constraint=False)
 
     # The integer ID of the app inside App Store Connect.
-    #
-    # TODO: On the AppStoreConnect API this is a str field, it was a mistake to make this
-    #    int.
     app_id = models.IntegerField()
 
+    # The integer ID of the app inside App Store Connect, as a string.
+    app_id_str = models.CharField(max_length=256)
```
Comment (wedamija) [inline src/sentry/models/appconnectbuilds.py:34]: You'll need to include the default here. Currently it doesn't match the migration, so you're getting an error where it's detecting a difference between the model state and migration state.

File: src/sentry/migrations/0326_add_appid_str.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    dependencies = [
+        ("sentry", "0325_turn_off_suspect_committer_auto_assignment_as_default"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunSQL(
+                    """
+                    ALTER TABLE "sentry_appconnectbuild" ADD COLUMN "app_id_str" character(256) NOT NULL DEFAULT '0';
```
Comment (wedamija) [inline src/sentry/migrations/0326_add_appid_str.py:35]: I think you'll want this to be a `varchar(256)`. `character` pads out the rest of the column with spaces, which I don't think is what you want.

**Review by wedamija — APPROVED** [review]

**Pull request merged**